docs/ocis/adr/0006-service-discovery.md

```diff
@@ -0,0 +1,58 @@
+---
+title: "6. Service Discovery within oCIS and Reva"
+date: 2021-04-19T13:00:00+01:00
+geekdocRepo: https://github.com/owncloud/ocis
+geekdocEditPath: edit/master/docs/ocis/adr
+geekdocFilePath: 0006-service-discovery.md
+---
+
+# 6. Service Discovery within oCIS and Reva
+
+* Status: proposed
+* Deciders: @refs, @butonic, @micbar, @dragotin, @pmaier1
+* Date: 2021-04-19
+
+Technical Story: [Introduce Named Services.](https://github.com/cs3org/reva/pull/1509)
+
+## Context and Problem Statement
+
+Reva relies heavily on config files. Implications of this approach are having to know a-priori where a service is running (host + port). We want to move away from hardcoded values and rely instead on named services for service discovery. Furthermore, we would like both platforms (Reva + oCIS) to have the same source of truth at any given time, not having one to notify the other whenever a service status changes.
+
+## Decision Drivers
+
+* Avoid a-priori knowledge of services.
+* Ease of scalability.
+* Always up-to-date knowledge of the running services on a given deployment (a service registry doesn't have to necessarily be running on the same machine / network)
+
+## Considered Options
+
+* Hardcoded tuples of hostname + port
+* Dynamic service registration
+
+## Decision Outcome
+
+Chosen option: "Dynamic service registration". There were some drawbacks regarding this due to introducing go-micro to Reva was from start a problem, given how much of a bloat it would mean given the little usage of it. Instead we decided to define our very own [Registry interface](https://github.com/refs/reva/blob/58d013a7509d1941834e1bc814e9a9fa8bff00b1/pkg/registry/registry.go#L22-L35) on Reva and extended the runtime arguments to [allow for injecting a registry](https://github.com/refs/reva/blob/58d013a7509d1941834e1bc814e9a9fa8bff00b1/cmd/revad/runtime/option.go#L53-L58).
```

Comment:
Could split in two sentences

1) There were some drawbacks...
2) Introducing go-micro to Reva...